### ISSUE TITLE
[FEAT-REQ] profile/account support

### ISSUE THREAD

**Issue by terryrankine** [trek10inc/awsets#8]
Hi

the code currently supports AWS_PROFILE env var, do you think you could like that back to the name of the cache file, or smuggle account number into there somewhere?

I would love to have a cache of a few of the clients i work on, at the moment I have to do a little bit of files/folder manipulation to add this 'external' metadata.

Open to your thoughts.

Terry

**Comment by jcarter3** [comment]
Hey! The cache is meant to segment the results by account already:
https://github.com/trek10inc/awsets/blob/master/cmd/awsets/cache/cache.go#L32
For each account Id, it creates/uses a different bucket within the boltdb database. Are you seeing situations where this is not working as expected?

**Comment by terryrankine** [comment]
so:
* have an 'old aws account' and a 'new one'
* I am moving stuff from one to another over time
* i want to check using your tool that its close to the same
* i want the best way to run the same thing on each account and make a pic

do i:
1. run it in different dirs - named after the account/role im using
2. run it in the same dir and just store the output in different files? what happens with the cache?
3. some combo i havent thought about?

how long does the cache last? how can you tell if its expired? is there an option to force use old cache? as opposed to force update?

I feel that i have blured the lines on this issue - and if this should be a forum question - happy to move it 👍 
Thanks

**Comment by jcarter3** [comment]
* The cache will save off by account id. If you want to compare the files, just name the output differently for each account.
* The cache is currently set at 1 day, but you can pass a flag `--refresh` to ignore it - no option to ignore the time limit though. There isn't really a good way to tell if it's expired other than enable verbose logging - cache misses are logged (but very noisy). I'd like to add in better caching at some point, so input is welcome

What this would <currently> look like until the `--profile` flag is complete:
```
AWS_PROFILE=old_acct awsets list -o old_acct.json
AWS_PROFILE=new_acct awsets list -o new_acct.json
```
The image that is generated is different each time, so that wouldn't currently be a good way to compare the accounts. Unless there is a deterministic dot render that could be used... Otherwise I'd probably just throw together a quick script to compare the outputs based on various properties. A `diff` would likely be too noisy since timestamps and any automatic AWS naming would show up different between the two.

**Comment by terryrankine** [comment]
You rock – can you add that snippet to your readme in GIT – I think its worth it.
For me, Visual diffs on an image will show evolution over time, and be helpful enough to ask detailed questions

Thanks

[closed]

**Comment by jcarter3** [comment]
Support for a `--profile` flag is now included in the `v0.4.0` release. I'll close this in a few days if there aren't any other issues around this. Thanks!
